src/main/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceService.scala

```diff
@@ -1,107 +1,65 @@
-      case Left(error) =>
-        log.error("Unable to unmarshal entity -- " + error.toString)
-        entity
-    }
   }
 
-  val routes: Route =
-    pathPrefix("workspaces") {
-      pathEnd {
-        mapHttpResponseEntity(transformListWorkspaceRequest) {
-          passthrough(rawlsWorkspacesRoot, HttpMethods.GET)
-        } ~
-        post {
-          entity(as[WorkspaceCreate]) { createRequest => requestContext =>
-            val extReq = Post(FireCloudConfig.Rawls.workspacesUrl, new RawlsWorkspaceCreate(createRequest))
-            externalHttpPerRequest(requestContext, extReq)
-          }
-        }
-      } ~
-      pathPrefix(Segment / Segment) { (workspaceNamespace, workspaceName) =>
-        val workspacePath = rawlsWorkspacesRoot + "/%s/%s".format(workspaceNamespace, workspaceName)
-        pathEnd {
-          mapHttpResponseEntity(transformSingleWorkspaceRequest) {
-            passthrough(workspacePath, HttpMethods.GET)
-          } ~
-          passthrough(workspacePath, HttpMethods.DELETE)
-        } ~
-        path("methodconfigs") {
-          passthrough(workspacePath + "/methodconfigs", HttpMethods.GET, HttpMethods.POST)
-        } ~
-        path("importEntities") {
-          post {
-            formFields( 'entities ) { entitiesTSV =>
-              respondWithJSON { requestContext =>
-                perRequest(requestContext, Props(new EntityClient(requestContext)),
-                  EntityClient.ImportEntitiesFromTSV(workspaceNamespace, workspaceName, entitiesTSV))
-              }
-            }
-          }
-        } ~
-        path("updateAttributes") {
-          passthrough(workspacePath, HttpMethods.PATCH)
-        } ~
-        path("acl") {
-          passthrough(workspacePath + "/acl", HttpMethods.GET, HttpMethods.PATCH)
-        } ~
-        path("checkBucketReadAccess") {
-          passthrough(workspacePath + "/checkBucketReadAccess", HttpMethods.GET)
-        } ~
-        path("clone") {
-          post {
-            entity(as[WorkspaceCreate]) { createRequest => requestContext =>
-              val extReq = Post(workspacePath + "/clone", new RawlsWorkspaceCreate(createRequest))
-              externalHttpPerRequest(requestContext, extReq)
-            }
-          }
-        } ~
-        path ("lock") {
-          passthrough(workspacePath + "/lock", HttpMethods.PUT)
-        } ~
-        path ("unlock") {
-          passthrough(workspacePath + "/unlock", HttpMethods.PUT)
-        }
+  def updateWorkspaceACL(workspaceNamespace: String, workspaceName: String, aclUpdates: List[WorkspaceACLUpdate], originEmail: String) = {
+
+    val aclUpdate = rawlsDAO.patchWorkspaceACL(workspaceNamespace, workspaceName, aclUpdates)
+    aclUpdate flatMap { actualUpdates =>
+
+      val (addedNotifications, removedNotifications) = actualUpdates.partition(!_.accessLevel.equals(WorkspaceAccessLevels.NoAccess)) match { case (added, removed) =>
```

Comment:
why not remove the `!` in the partition and flip the order of `(added, removedNotifications)`